tc/core/polyhedral/schedule_tree.cc

```diff
@@ -156,6 +156,8 @@ static std::unique_ptr<ScheduleTreeElemBase> makeElem(const ScheduleTree& st) {
 ////////////////////////////////////////////////////////////////////////////////
 ScheduleTree::ScheduleTree(isl::ctx ctx) : ctx_(ctx) {}
 
+ScheduleTree::~ScheduleTree() {}
```

Comment:
What is the point of having 3 commits for introducing virtual methods?
It creates an intermediate state in which methods have virtual methods but no virtual destructors.
In any case I think the splitting of commits goes a bit too far here, I'd squash the 3 commits in 1.

Reply:
That can be fixed by swapping the order of commits